homeassistant/components/influxdb/sensor.py

```diff
@@ -163,13 +239,28 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
                 entities.append(InfluxSensor(hass, influx, query))
             else:
                 _LOGGER.error(NO_BUCKET_ERROR, query[CONF_BUCKET])
-    else:
+
+    if CONF_QUERIES in config:
         for query in config[CONF_QUERIES]:
             if query[CONF_DB_NAME] in influx.data_repositories:
                 entities.append(InfluxSensor(hass, influx, query))
             else:
                 _LOGGER.error(NO_DATABASE_ERROR, query[CONF_DB_NAME])
 
+    if CONF_QUERIES_RAW in config:
+        for query in config[CONF_QUERIES_RAW]:
+            if query[CONF_API_VERSION] == DEFAULT_API_VERSION:
+                if query[CONF_DB_NAME] in influx.data_repositories:
+                    entities.append(InfluxSensor(hass, influx, query))
+                else:
+                    _LOGGER.error(NO_DATABASE_ERROR, query[CONF_DB_NAME])
+
+            if query[CONF_API_VERSION] == API_VERSION_2:
+                if query[CONF_BUCKET] in influx.data_repositories:
+                    entities.append(InfluxSensor(hass, influx, query))
+                else:
+                    _LOGGER.error(NO_BUCKET_ERROR, query[CONF_BUCKET])
+
```

Comment:
```suggestion
```
Remove this for the same reasons as discussed in other comments. We shouldn't prompt for a bucket with V2, it just goes in the raw flux query.

Reply:
Ok, will be removed